build.sbt

```diff
@@ -68,6 +69,7 @@ lazy val `core-common` = crossProject(JVMPlatform, JSPlatform, NativePlatform)
     name := "otel4s-core-common",
     libraryDependencies ++= Seq(
       "org.typelevel" %%% "cats-core" % CatsVersion,
+      "org.typelevel" %%% "vault" % VaultVersion,
```

Comment:
From the spec:

> Languages are expected to use the single, widely used Context implementation if one exists for them.

They're obviously thinking of Go there.  But Vault meets all the requirements, and is a stable and proven Typelevel component.  Interpreting somewhat liberally, Vault is our obvious context?